### PR TITLE
[MRG] add option to do cubic spline interpolation

### PR DESCRIPTION
**Pull request by jasmainak**
It seems that cubic spline interpolation is how stim artifacts are usually fixed.

Implemented by @padmasundaram. I just added a test and made the PR

### CODE REVIEW

File: mne/preprocessing/stim.py [hunk 1]
```diff
@@ -23,13 +23,20 @@ def _get_window(start, end):
...
     if mode == 'linear':
         x = np.array([first_samp, last_samp])
         f = interp1d(x, data[:, (first_samp, last_samp)][picks])
         xnew = np.arange(first_samp, last_samp)
         interp_data = f(xnew)
         data[picks, first_samp:last_samp] = interp_data
+    if mode == 'cubic':
+        x = np.array([first_samp, last_samp])
+        for pick in picks:
+            f = PchipInterpolator(x, data[:, (first_samp, last_samp)][pick])
```
Comment (Eric89GXL) [inline mne/preprocessing/stim.py]: why not just `data[pick][x]`? It copies less data and is simpler

File: mne/preprocessing/stim.py [hunk 2]
```diff
@@ -55,9 +62,10 @@ def fix_stim_artifact(inst, events=None, event_id=None, tmin=0.,
         Start time of the interpolation window in seconds.
     tmax : float
         End time of the interpolation window in seconds.
-    mode : 'linear' | 'window'
+    mode : 'linear' | 'cubic' | 'window'
         Way to fill the artifacted time interval.
         'linear' does linear interpolation
+        'cubic' does cubic spline interpolation
```
Comment (Eric89GXL) [inline mne/preprocessing/stim.py]: monotonic cubic, and you might as well put in parens:
```
:func:`scipy.interpolate.PchipInterpolator`
```

File: mne/preprocessing/tests/test_stim.py [hunk 3]
```diff
@@ -39,11 +39,12 @@ def test_fix_stim_artifact():
...
-    data = epochs.get_data()[:, :, tmin_samp:tmax_samp]
-    diff_data0 = np.diff(data[0][0])
-    diff_data0 -= np.mean(diff_data0)
-    assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
+    for mode in ['linear', 'cubic']:
+        epochs = fix_stim_artifact(epochs, tmin=tmin, tmax=tmax, mode=mode)
+        data = epochs.get_data()[:, :, tmin_samp:tmax_samp]
+        diff_data0 = np.diff(data[0][0])
+        diff_data0 -= np.mean(diff_data0)
+        assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))
```
Comment (agramfort) [inline mne/preprocessing/tests/test_stim.py]: this diff makes sense for linear interp. If this passes it means cubic interp does a linear interp in the end.

**Comment by jasmainak** [comment]
I'm closing this PR for now. Not convinced that cubic interpolation offers anything more than linear interpolation. I'll reopen if something comes up later ...

**Pull request closed without merging**